quinn-proto/src/connection/mod.rs

```diff
@@ -386,7 +389,18 @@ where
                 // Tail loss probes must not be blocked by congestion, or a deadlock could arise
                 if ack_eliciting
                     && self.spaces[space_id].loss_probes == 0
-                    && self.congestion_blocked()
+                    && (self.congestion_blocked() || {
```

Comment:
Let's just do this as a separate `if` statement? This is a bit too deeply nested for my taste.